src/main/java/com/tabnine/binary/BinaryProcessRequesterProvider.java

```diff
@@ -80,9 +85,11 @@ private void createNew() {
         }
         BinaryProcessGateway binaryProcessGateway = binaryProcessGatewayProvider.generateBinaryProcessGateway();
 
+        this.binaryProcessRequester = new BinaryProcessRequesterImpl(new ParsedBinaryIO(new GsonBuilder().create(), binaryProcessGateway));
```

Comment:
this could lead to an unexpected behavior , `executeRequest` is not synchronized, meaning `createNew` can be called while `initProcess` is still running, and because `binaryProcessRequester` is inited before the `initProcess` it will always be destroyed on line 84

Reply:
moved back to original location and the poller will create its own instance